File: 1542-consecutive-characters/1542-consecutive-characters.py
class Solution:
    def maxPower(self, s: str) -> int:
        curr = 1 
        maxp = 1  
        for i in range(1, len(s)):
            if s[i] == s[i - 1]:
                curr += 1
            else:
                curr = 1
            maxp = max(maxp, curr)

        return maxp

File: 1542-consecutive-characters/1542-consecutive-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var maxPower = function(s) {
    let curr = 1;
    let maxp = 1;
    for (let i = 1; i < s.length; i++) {
        if (s[i] === s[i - 1]) {
            curr++;
        } else {
            curr = 1;
        }
        maxp = Math.max(maxp, curr);
    }

    return maxp;
};
